docs/proposals/environment-variables.md

```diff
@@ -0,0 +1,66 @@
+# Proposal: OS Environment Variables related Operations
+
+_Owners_: @daneshk @MadhukaHarith92  
+_Reviewers_: @daneshk  
+_Created_: 2022/12/18   
+_Updated_: 2022/12/18  
+_Issues_: [#2666](https://github.com/ballerina-platform/ballerina-standard-library/issues/2666)  [#2630](https://github.com/ballerina-platform/ballerina-standard-library/issues/2630)
+
+## Summary
+At the moment, the Ballerina OS module only has APIs to retrieve environment variable information. However, it’s not possible to perform other operations such as setting new environment variables, listing them, and unsettling particular environment variables. In this proposal, we describe how these requirements can be fascilitated.
+
+## Goals
+To support setting, unsettling, and listing environment variables.
+
+## Motivation
+To allow users to manipulate environment variables programmatically.
+
+## Description
+
+```ballerina
+# Represents OS module related errors.
+public type Error distinct error;
+```
+
+- The `setEnv` function can be used to set any environment variable programmatically.
+
+```ballerina
+# Sets the value of the environment variable named by the key. 
+# ```ballerina
+# os:Error err = os:setEnv("BALCONFIGFILE", "/path/to/Config.toml");
+# ```
+#
+# + key - Key of the environment variable
+# + value - Value of the environment variable
+# + return - error if setting the environment variable fails
+public isolated function setEnv(string key, string value) returns Error;
+```
+
+- Users can remove any particular environment variable from the system using the `unsetEnv` function.
```

Comment:
Better we explain what we do in negative scenarios like,  we try to unset nonexistence environment property

Reply:
Updated in [2033edc](https://github.com/ballerina-platform/module-ballerina-os/pull/415/commits/2033edcba3c56ca7f69271d3d3519c8ff388fd64)